plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -71,14 +71,14 @@ def verify
 
     def build_selectors
       all = containers + init_containers
-      all.each_with_index.map { |c, i| build_selector_for_container(c, first: i == 0) }.compact
+      all.each_with_index.map { |c, i| build_selector_for_container(c, index: i) }.compact
     end
 
     private
 
-    def build_selector_for_container(container, first:)
-      dockerfile = container[:"samson/dockerfile"] ||
-        (first && ENV['KUBERNETES_ADDITIONAL_CONTAINERS_WITHOUT_DOCKERFILE'] ? 'none' : 'Dockerfile')
+    def build_selector_for_container(container, index:)
+      dockerfile = samson_container_config(container, :"samson/dockerfile") ||
+        (index == 0 && ENV['KUBERNETES_ADDITIONAL_CONTAINERS_WITHOUT_DOCKERFILE'] ? 'none' : 'Dockerfile')
```

Comment:
I analysed and tested this code and came to the conclusion that this code should be `index != 0` instead.

We want the `dockerfile` variable to be `'none'` only with the sidecar containers, don't we? So they don't get their image tag replaced. It's the first one the only one that has to have the hash inserted. This way it happens the other way round.